### DISCUSSION TITLE
Bearer Token

### DISCUSSION THREAD

**Discussion by zxdd7525** [netbox-community/pynetbox#607] [Ideas]
Hello,
For **Oauth2** authorization, we need to pass a token type **Bearer** instead of **Token**.

Would it be possible to add an argument 'token_type' that will be 'Token' by default, that permit to change the token type used in http headers.

The changes to make seem simple.
In the file /pynetbox/core/query.py 

1 - add an argument token_type (with default value to **Token**)
```
 def __init__(
   ...
   token_type="Token",
   ...
   ):
  self.token_type = token_type
```
2 - change all lines (lines 208 & 232)
```
 headers["authorization"] = "Token {}".format(self.token)
```
with 
```
 headers["authorization"] = f"{} {}".format(self.token_type, self.token)
```

A check could also be done for the type of token given (should be Token or Bearer ).

For information, I also ask to make this change to Ansible netbox module. The changes have been implemented.

https://github.com/netbox-community/ansible_modules/blob/devel/plugins/inventory/nb_inventory.py

Thanks a lot,

P. Morry

**Comment by abhi1693** [comment]
Where in netbox a Bearer used? I did a quick code search and nothing came up

**Reply by YMourache** [reply]
for example when Netbox instance is placed behind an authentication proxy that requires a Bearer Token in the Head of the Request, we cannot use pynetbox for that purpose unless we override the session object with a custom code...

**Comment by zxdd7525** [comment]
It's not native to netbox, a plugins have to be added to use a bearer token with netbox (In our organisation, we added it).
But for all other organisation that want to use Oauth2 with netbox, they will have to patch pynetbox too.

**Reply by abhi1693** [reply]
Currently, I don't feel comfortable to add this for 1 specific use case unless we get more requests

**Reply by zxdd7525** [reply]
As said, we use it often in our organization ( big one), and if people don't have this feature included, they have to implemente the patch when there don't find an existing patch.
It'll be very appreciate to make this little change instead of creating a patch that is not so easy to create when you're not a developer with experience.
